### PR TITLE
Adding EitherT run aws operations

### PR DESCRIPTION
**Pull request by nhibberd**
@charleso

### CODE REVIEW

**Comment by charleso** [comment]
:+1: Although would prefer you call the IO versions and then just catch/return rather than duplicate the internal logic.

File: mismi-core/src/Mismi/Control.hs [hunk 1]
```diff
@@ -43,6 +48,14 @@ runAWS e =
   let e' = over envManager (\m -> m { mResponseTimeout = Just 60000000 }) e
   in runResourceT . A.runAWS e'
 
+runAWST :: (MonadIO m, MonadCatch m) => Env -> AWS a -> EitherT Error m a
+runAWST e =
+  catchError . runAWS e
+
+catchError :: (MonadIO m, MonadCatch m) => IO a -> EitherT Error m a
```
Comment (charleso) [inline mismi-core/src/Mismi/Control.hs]: Expose this too?

**Comment by charleso** [comment]
:+1:

File: mismi-core/src/Mismi/Control.hs (hunk 1 above)
Comment (nhibberd) [inline mismi-core/src/Mismi/Control.hs]: Unsure tbh

File: mismi-core/src/Mismi/Control.hs [hunk 2]
```diff
@@ -43,6 +48,14 @@ runAWS e =
...
+catchError action =
+  (EitherT . try . liftIO $ action) `catch` \(err :: Error) -> left err
```
Comment (markhibberd) [inline mismi-core/src/Mismi/Control.hs]: Is that catch doing anything? Doesn't try capture this case?

Comment (EduardSergeev) [inline mismi-core/src/Mismi/Control.hs]: I also was under impression that `EitherT . try` is all we need here.

**Pull request merged**